### PR TITLE
Use new Vega `layout`

### PR DESCRIPTION
**Pull request by kanitw**
- Rename `assembleLayout()` to `assembleLayoutData()` and add assembleLayout() for really assembling the new `layout` object.

- We can re-use this infrastructure for all of `facet`, `concat`, and `repeat`.

- We can make this branch the new merge target until we have a new Vega release.  (As after we add layout, Travis won't pass until then.)

## TODO

- [ ] Bring `main-group` back for necessary cases
- [ ] Test if interactive specs are still working
- [x] Wait for Vega release

### CODE REVIEW

File: src/compile/facet.ts [hunk 1]
```diff
@@ -256,9 +257,14 @@ export class FacetModel extends Model {
...
+  public assembleLayout(): VgLayout {
+    // TODO: add layout
+    return null;
+  }
+
+  public assembleLayoutData(layoutData: VgData[]): VgData[] {
     // Postfix traversal – layout is assembled bottom-up
-    this.child.assembleLayout(layoutData);
+    this.child.assembleLayoutData(layoutData);
     return assembleLayout(this, layoutData);
```
Comment (domoritz) [inline src/compile/facet.ts]: rename?

**Comment by domoritz** [comment]
LGTM

**Comment by domoritz** [comment]
So now we are only blocked by a vega beta 29 release, wohoo.

**Comment by kanitw** [comment]
And the main-group issue that i will take a stab tmr

**Pull request merged**